amethyst_rendy/src/submodules/lut.rs

```diff
@@ -0,0 +1,8451 @@
+            binding,
+            handle: handle.clone(),
+            layout,
+        })
+    }
+
+    /// Try to insert a new texture for submission in this texture batch.
+    pub fn insert(
+        &mut self,
+        factory: &Factory<B>,
+        world: &World,
+        handle: &Handle<Texture>,
+        layout: hal::image::Layout,
+        binding: u32
+    ) -> Option<bool> {
+        #[cfg(feature = "profiler")]
+        profile_scope!("insert");
+
+        match self.luts.get(&binding) {
+            Some(LutState::Loaded { .. }) => {
+                return Some(false);
+            }
+            _ => {}
+        };
+
+        let (new_state, loaded) = self
+            .try_insert(factory, handle, layout, binding)
+            .map(|s| (s, true))
+            .unwrap_or_else(|| {
+                (
+                    LutState::Unloaded{ marker: PhantomData},
+                    false,
+                )
+            });
+        self.luts.insert(binding, new_state);
+
+
+        if loaded {
+            Some(true)
+        } else {
+            None
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct LUTSub<B: Backend> {
+    layout: RendyHandle<DescriptorSetLayout<B>>,
+    luts: HashMap<u32, LutState< B>>,
+    set: Escape<DescriptorSet<B>>
+}
+impl<B: Backend> LUTSub<B> {
+
+    /// Returns the raw `DescriptorSetLayout` for a Texture
+    pub fn raw_layout(&self) -> &B::DescriptorSetLayout {
+        self.layout.raw()
+    }
+    
+    pub fn maintain(&mut self, factory: &Factory<B>, world: &World) {
+        #[cfg(feature = "profiler")]
+        profile_scope!("maintain");
+
+
+        let mut sets = Vec::new();
+        use crate::util::texture_desc;
+        use crate::util::desc_write;
+        let tex_storage = <(Read<'_, AssetStorage<Texture>>)>::fetch(world);
+
+        
+        
+        for (&binding, lut) in self.luts.iter() {
+            if let LutState::Loaded { handle, layout, .. } = lut {
+                let tex = tex_storage.get(&handle);
+                
+                if let Some(tex) = tex {
+                    if let Some(desc) = texture_desc(tex, *layout) {
+                        unsafe {
+                            let set = self.set.raw();
+                            sets.push(desc_write(set, binding, desc));
+                        }
+                    }
+                }
+            }
+        }
+        unsafe {
+            factory.write_descriptor_sets(sets);
+        }
+    }
+
+    /// Bind all textures
+    #[inline]
+    pub fn bind(
+        &self,
+        pipeline_layout: &B::PipelineLayout,
+        set_id: u32,
+        encoder: &mut RenderPassEncoder<'_, B>,
+    ) {
+        unsafe {
+            encoder.bind_graphics_descriptor_sets(
+                pipeline_layout,
+                set_id,
+                Some(self.set.raw()),
+                std::iter::empty(),
+            );
+        }
+           
+    }
+
+    
+}
+lazy_static! {
```

Comment:
Can this get moved out to a seperate file? either via `include!` or some other means. I love the idea of having static LUTs we can define - but we should probably separate the data out from the submodule definition.

Reply:
Yeah. As this is still WIP this will definitely change. Image loading support is missing too for LUTs. The included luts will probably loaded as .dds files if I get dds support added somewhere in a timely manner.